src/computer_player.c

```diff
@@ -0,0 +1,604 @@
+{
+    struct DECK *ptr = *handcard;
+    while (ptr->next != NULL)
+    {
+        if (ptr->card.name == inputCardName && ptr->card.color == inputCardColor)
+        {
+            return 1;
+        }
+        ptr = ptr->next;
+    }
+    return 0;
+}
+
+
+/*
+* @brief: function that find the largest number of given color
+*
+* @return CardName the largest number
+*/
+
+enum CardName find_largest_number(struct DECK **handcard, enum CardColor inputCardColor)
+{
+    struct DECK *temp = *handcard;
+    enum CardName Max = ZERO;
+
+
+    // Check loop while head not equal to NULL
+    while (temp != NULL)
+    {
+        
+        // If max is less then head->data then
+        // assign value of head->data to max
+        // otherwise node point to next node.
+        if (temp->card.color == inputCardColor && Max < temp->card.name)
+        {
+            Max = temp->card.name;
+        }
+        temp = temp->next;
+    }
+    
+    return Max;
+}
+
+/*
+* @brief: function that find the occruence of a number
+*
+* @return int number count for occruence
+*/
+
+int find_occurence_of_number(struct DECK **handcard, enum CardName inputCardNumber)
+{
+    struct DECK *temp = *handcard;
+    int count = 0;
+    while (temp != NULL)
+    {
+        if (temp->card.name == inputCardNumber)
+        {
+            count++;
+        }
+        temp = temp->next;
+    }
+    free(temp);
+    return count;
+}
+
+/*
+* @brief: function that find the most color occruence of a given number
+*
+* @return CardColor  most color occruence of a given number
+*
+*/
+
+enum CardColor find_color_with_most_occurence(struct DECK **handcard, enum CardName inputCardNumber)
+{
+    struct DECK *temp = *handcard;
+    int R_count=0, B_count=0, Y_count=0, G_count=0;
+    while (temp != NULL)
+    {
+        if (temp->card.name == inputCardNumber)
+        {
+            if (temp->card.color == RED)
+            {
+                R_count++;
+            }
+            else if (temp->card.color == BLUE)
+            {
+                B_count++;
+            }
+            else if (temp->card.color == YELLOW)
+            {
+                Y_count++;
+            }
+            else
+            {
+                G_count++;
+            }
+        }
+        temp = temp->next;
+
+    }
+    if (R_count > B_count && R_count > G_count && R_count > Y_count)
+    {
+        return RED;
+    }
+    else if (B_count > R_count && B_count > G_count && B_count > Y_count)
+    {
+        return BLUE;
+    }
+    else if (G_count > R_count && G_count > B_count && G_count > Y_count)
+    {
+        return GREEN;
+    }
+    else if (Y_count > R_count && Y_count > B_count && Y_count > G_count)
+    {
+        return YELLOW;
+    }
+    return 0;
+ }
+
+
+/*
+* @brief: function that add one card at the end of the handcard list 
+*
+* @return void
+*
+*/
+void push(struct DECK** head, struct CARD card) {
```

Comment:
We have defined a common function: int add_card_at_end(Deck_t *p_head, Card_t card) in cards_management.c. Please call that function instead, and remove this redundant function here.